### DISCUSSION TITLE
Hide everything except

### DISCUSSION THREAD

**Discussion by mezuky04** [drakkan/sftpgo#1323] [General]
Hello. I have three directories: `shared-files`, `shared-videos` and `shared-documents`. Besides these three directories, there can be n other directories. I want to be able to configure any user to only see (and access) one or multiple directories and hide the other ones. For example, user1 should be able to see and access only `shared-files` and `shared-videos` while user2 can access `shared-files`, `shared-videos` and `shared-documents` and the other directories should not be visible (or accessible).

I tried to use per-directory pattern restrictions, but I'm not able to configure "hide everything except". 
<img width="1401" alt="image" src="https://github.com/drakkan/sftpgo/assets/7033003/4df3a460-4064-4b83-a3e2-ff8edefe11e3">

Any idea if I'm doing something wrong or extended globbing is not supported?

Thanks 🙂

**Comment by drakkan** [comment]
Hello,

the supported glob syntax is documented [here](https://pkg.go.dev/path#Match).
Please try something like this

![Schermata del 2023-06-14 14-29-56](https://github.com/drakkan/sftpgo/assets/553263/46d7c37e-b956-49a9-a18a-3110f7d25a74)

using the current development version. It should work, a confirmation would be greatly appreciated, thank you

**Comment by mezuky04** [comment]
Thanks for the quick response. Using something like `shared-*` will not work for my case because I need an exact match on the directory/file name. 

I tried to test your approach on a clean installation and something strange happens: `Hidden` is always set for the first row. I tried changing it to `Visible` and saving the changes, but `Hidden` is displayed as a value after reloading the page. I attached a screenshot.
<img width="1399" alt="image" src="https://github.com/drakkan/sftpgo/assets/7033003/63f7c5bc-39fe-4d5e-9020-e60e6cb729d7">

I'm using `SFTPGo 2.5.1-dev-7245710b-2023-06-12T18:12:08Z +metrics +azblob +gcs +s3 +bolt +mysql +pgsql +sqlite -unixcrypt +portable` which should be the last dev release, right?

**Reply by drakkan** [reply]
Hidden will be ignored for allowed paths, I'll update the web UI to show visible and avoid confusion.
If you need an exact match you have to add each allowed dir instead of `shared-*`

**Reply by mezuky04** [reply]
The hidden value confused me but you're right, it works as expected. I tried different patterns and had no problems. Any idea when this dev version will be released as stable?

**Reply by drakkan** [reply]
v2.5.2 has been released

**Comment by drakkan** [comment]
> The hidden value confused me but you're right, it works as expected. I tried different patterns and had no problems. Any idea when this dev version will be released as stable?

thanks for confirming. I'll fix the UI to avoid confusion. I plan to tag v2.5.2 next weekend or sooner
